eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Ratify.hs

```diff
@@ -135,6 +142,53 @@ accepted RatifyEnv {reStakePoolDistr = PoolDistr poolDistr} gas =
       HardForkInitiation {} -> 101 % 100
       _ -> spoThreshold
 
+dRepAccepted :: forall era. RatifyEnv era -> GovActionState era -> Ratio Word64 -> Bool
+dRepAccepted RatifyEnv {reDRepDistr} GovActionState {gasDRepVotes, gasAction} threshold =
+  case dRepAcceptedRatio reDRepDistr gasDRepVotes gasAction of
+    Nothing
+      | threshold == 0 -> True
+      | otherwise -> False
+    Just ratio -> ratio >= threshold
+
+-- Compute the dRep ratio yes/(yes + no), where
+-- yes: is the total stake of
+--    - registered dReps that voted 'yes', plus
+--    - the AlwaysNoConfidence dRep, in case the action is NoConfidence
+-- no: is the total stake of
+--    - registered dReps that voted 'no', plus
+--    - registered dReps that did not vote for this action, plus
+--    - the AlwaysNoConfidence dRep
+-- In other words, the denominator `yes + no` is the total stake of all registered dReps, minus the abstain votes stake
+-- (both credential DReps and AlwaysAbstain)
+--
+-- We iterate over the dRep distribution, and incrementally construct the numerator and denominator.
+dRepAcceptedRatio ::
+  forall era.
+  Map (DRep (EraCrypto era)) (CompactForm Coin) ->
+  Map (Credential 'DRepRole (EraCrypto era)) Vote ->
+  GovAction era ->
+  Maybe (Ratio Word64)
```

Comment:
Ratio should never be used with fixed width integral numbers. They are susceptible to overflows in very surprising ways. See my comments on this topic in this issue: https://github.com/input-output-hk/cardano-ledger/pull/2326
```suggestion
  Maybe (Ratio Integer)
```